src/features/preferences/renderer/preference-items/terminal/terminal-font-family/terminal-font-options.injectable.tsx

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import type { IComputedValue } from "mobx";
+import { action, computed } from "mobx";
+import React from "react";
+import type { SingleValue } from "react-select";
+import userStoreInjectable from "../../../../../../common/user-store/user-store.injectable";
+import { defaultTerminalFontFamily } from "../../../../../../common/vars";
+import type { SelectOption } from "../../../../../../renderer/components/select";
+import { terminalFontInjectionToken } from "../../../../../terminal/renderer/fonts/token";
+
+export interface TerminalFontPreferenceModel {
+  readonly options: IComputedValue<SelectOption<string>[]>;
+  readonly current: IComputedValue<string>;
+  set: (selection: SingleValue<SelectOption<string>>) => void;
+}
+
+const terminalFontPreferenceModelInjectable = getInjectable({
+  id: "terminal-font-preference-model",
+  instantiate: (di): TerminalFontPreferenceModel => {
+    const userStore = di.inject(userStoreInjectable);
+    const terminalFonts = di.injectMany(terminalFontInjectionToken);
+
+    return {
+      options: computed(() => terminalFonts.map(font => ({
+        label: (
+          <span style={{ fontFamily: `${font.name}, var(--font-terminal)`, fontSize: userStore.terminalConfig.fontSize }}>
+            {font.name}
+          </span>
+        ),
+        value: font.name,
+        isSelected: userStore.terminalConfig.fontFamily === font.name,
+      }))),
+      current: computed(() => userStore.terminalConfig.fontFamily),
+      set: action(selection => {
+        userStore.terminalConfig.fontFamily = selection?.value ?? defaultTerminalFontFamily;
+      }),
```

Comment:
_nit:_ better readability + `set()` should not know anything about `SelectOption`, better to pass just a font value:

```suggestion
      fontFamily: computed(() => userStore.terminalConfig.fontFamily),
      setFont: action((font => {
        userStore.terminalConfig.fontFamily = font ?? defaultTerminalFontFamily;
      }),
```

Reply:
I can update the name of the functions sure but using `computed(() => ...)` better communicates the contract then a getter.

Reply:
Better for who? Why end-user of this let's say api should care about `computed` that on that side or not?

Reply:
In other words to distinguish `observable` this field or not?

Reply:
> Why end-user of this let's say api should care about computed that on that side or not?

So that the API user knows that they should only read the field in an observable context or that reading it multiple times outside of a context may result in different values.